core/src/integrationTest/java/amino/run/multidm/MultiDMTestCases.java

```diff
@@ -67,16 +72,52 @@ private void runTest(String... dmNames) throws Exception {
             microServiceId = registry.create(spec.toString());
             KVStore store = (KVStore) registry.acquireStub(microServiceId);
 
-            // consensus DM needs some time to elect the leader other wise function call will fail
-            if (spec.getName().contains(CONSENSUS)) {
-                Thread.sleep(5000);
-            }
-
             for (int i = 0; i < 10; i++) {
                 String key = "k1_" + i;
                 String value = "v1_" + i;
-                store.set(key, value);
-                String returnValue = (String) store.get(key);
+                long startTime = System.currentTimeMillis();
+                String returnValue = "";
+                while (System.currentTimeMillis() - startTime < retryTimeoutMs) {
+                    try {
+                        store.set(key, value);
+                        returnValue = (String) store.get(key);
+                        System.out.println("Got value " + returnValue + " for key " + key);
+                        if (value.equals(returnValue)) {
+                            break; // Success, no more retries necessary
+                        }
+                    } catch (RuntimeException r) {
+                        System.out.println(
+                                "Runtime exception after "
+                                        + (System.currentTimeMillis() - startTime)
+                                        + "ms for key "
+                                        + key
+                                        + ", value "
+                                        + value
+                                        + " : "
+                                        + r.toString());
+                        if (r.getCause() != null
+                                && r.getCause()
+                                        .getClass()
+                                        .isAssignableFrom(LeaderException.class)) {
+                            System.out.println("Cause of runtime exception is LeaderException");
+                            if (!Arrays.asList(dmNames).contains(AT_LEAST_ONCE_RPC)) {
+                                // Swallow the exception and retry, after sleeping
+                                System.out.println(
+                                        "Swallowing runtime exception because AtLeastOnceRPC is not used.");
+                                Thread.sleep(retryPeriodMs);
+                            } else {
+                                System.out.println(
+                                        "Not swallowing runtime exception because AtLeastOnceRPC is used.");
```

Comment:
Seem to have swallowed here. right ? Same in below else too ?

Reply:
Oops!  Good catch.  I'll fix that.

Reply:
Fixed.

Reply:
Fixed.